python/ml4ir/model/metrics/metrics_helper.py

```diff
@@ -0,0 +1,69 @@
+import pandas as pd
+from typing import List
+
+
+def get_grouped_stats(
+    df: pd.DataFrame,
+    query_key_col: str,
+    label_col: str,
+    old_rank_col: str,
+    new_rank_col: str,
+    group_keys: List[str] = [],
+):
+    """Compute metrics on a data batch/chunk"""
+
+    # Select clicked records
+    df_clicked = df[df[label_col] == 1.0]
+
+    if group_keys:
```

Comment:
does this check if `group_keys` is empty?

Reply:
Yep.